lib/elixir/test/elixir/protocol_test.exs

```diff
@@ -143,11 +147,23 @@ defmodule ProtocolTest do
   end
 
   test "protocol defines callbacks" do
-    assert [{:type, 13, :fun, args}] = get_callbacks(@sample_binary, :ok, 1)
-    assert args == [{:type, 13, :product, [{:user_type, 13, :t, []}]}, {:type, 13, :boolean, []}]
+    assert [{:type, 17, :fun, args}] = get_callbacks(@sample_binary, :ok, 1)
+    assert args == [{:type, 17, :product, [{:user_type, 17, :t, []}]}, {:type, 17, :boolean, []}]
+
+    assert [{:type, 27, :fun, args}] = get_callbacks(@with_any_binary, :ok, 1)
+    assert args == [{:type, 27, :product, [{:user_type, 27, :t, []}]}, {:type, 27, :term, []}]
+  end
+
+  test "protocol with no functions is a behaviour" do
+    refute capture_err(fn ->
+             Code.eval_string("""
+             defprotocol ProtoNoFunctions do
+             end
```

Comment:
You can define the protocol directly. I believe the suite runs with --warnings-as-errors on CI, which means we would fail on those warnings. I tend to prefer that to `refute` because `refute` checks for a specific message (which may change in the future!).

```suggestion
    defprotocol ProtoNoFunctions do
    end
```

Reply:
That sounds much cleaner, thanks!